Game/Src/UI/HealthUI.gd

```diff
@@ -1,20 +1,25 @@
 extends Control
 
 onready var stats = $StatsGen
-
 onready var green =$Green
 onready var red = $Red
+onready var size = red.rect_size.x
+export var max_health = 100
+
+export(PackedScene) var statsScript
 
-var max_damage
+var health_unit
 
 func _ready():
-	max_damage = green.rect_scale.x
-	
-func set_damage(value):
-	print('health changed by ',value)
-	GlobalVariables.damage = clamp(value, 0, max_damage)
-	if red != null:
-		red.rect_scale.x = value
-		
+	health_unit = size / max_health
+	print('Health unit ',size,' ',health_unit,' ',max_health)
+	pass
 
+func change_red(value:float)->void:
+	print('changing red ',value,' ',value*health_unit)
```

Comment:
I think I understand that the health bar is actually composed of a red bar that is full and a green bar that varies in size and sits on top of it. When damages are taken, we rescale the green bar, making the red background bar visible. Is that correct?
So here we give the amount of damage in [health_units] and change the green bar? Then the function should be renamed to change_green()
also useless prints

Reply:
This is the other way around but true. Green is always full and red is cropped to cover the green bar with damage.

Reply:
Or I made a mistake in modifying Arjun's code ^^